Common/src/gRPC/EnumFieldExt.cs

```diff
@@ -0,0 +1,219 @@
+         TaskSummaryEnumField.ReceivedAt              => data => data.ReceptionDate,
+         TaskSummaryEnumField.AcquiredAt              => data => data.AcquisitionDate,
+         TaskSummaryEnumField.Error                   => data => data.Output.Error,
+         TaskSummaryEnumField.Unspecified             => throw new ArgumentOutOfRangeException(),
+         _                                            => throw new ArgumentOutOfRangeException(),
+       };
+
+
+  /// <summary>
+  ///   Convert gRPC field to <see cref="Expression" /> that represent how to access the field from the database object
+  /// </summary>
+  /// <param name="enumField">The gRPC message field</param>
+  /// <returns>
+  ///   The <see cref="Expression" /> that access the field from the object
+  /// </returns>
+  public static Expression<Func<TaskData, object?>> ToField(this TaskOptionEnumField enumField)
+    => enumField switch
+       {
+         TaskOptionEnumField.MaxDuration          => data => data.Options.MaxDuration,
+         TaskOptionEnumField.MaxRetries           => data => data.Options.MaxRetries,
+         TaskOptionEnumField.Priority             => data => data.Options.Priority,
+         TaskOptionEnumField.PartitionId          => data => data.Options.PartitionId,
+         TaskOptionEnumField.ApplicationName      => data => data.Options.ApplicationName,
+         TaskOptionEnumField.ApplicationVersion   => data => data.Options.ApplicationVersion,
+         TaskOptionEnumField.ApplicationNamespace => data => data.Options.ApplicationNamespace,
+         TaskOptionEnumField.ApplicationService   => data => data.Options.ApplicationService,
+         TaskOptionEnumField.EngineType           => data => data.Options.EngineType,
+         TaskOptionEnumField.Unspecified          => throw new ArgumentOutOfRangeException(),
+         _                                        => throw new ArgumentOutOfRangeException(),
+       };
+
+  /// <summary>
+  ///   Convert gRPC field to <see cref="Expression" /> that represent how to access the field from the database object
+  /// </summary>
+  /// <param name="enumField">The gRPC message field</param>
+  /// <returns>
+  ///   The <see cref="Expression" /> that access the field from the object
+  /// </returns>
+  public static Expression<Func<SessionData, object?>> ToField(this SessionRawEnumField enumField)
+    => enumField switch
+       {
+         SessionRawEnumField.SessionId    => session => session.SessionId,
+         SessionRawEnumField.Status       => session => session.Status,
+         SessionRawEnumField.PartitionIds => session => session.PartitionIds,
+         SessionRawEnumField.Options      => session => session.Options,
+         SessionRawEnumField.CreatedAt    => session => session.CreationDate,
+         SessionRawEnumField.CancelledAt  => session => session.CancellationDate,
+         SessionRawEnumField.Duration     => throw new ArgumentOutOfRangeException(),
```

Comment:
@aneojgurhem Why is SessionRawEnumField.Duration out of range? I mean what is the point of having duration in Api field but not in the corresponding field in Core?

Reply:
It is not implemented yet in the SessionData. We did not implement duration management